src/integrationTest/java/com/mongodb/kafka/connect/source/MongoSourceTaskIntegrationTest.java

```diff
@@ -579,6 +617,71 @@ void testCopyingExistingWithARestartMidwayThrough() {
     }
   }
 
+  @Test
```

Comment:
> So the integration tests are failing for version 3.6 which I'm assuming has something to do with changestream/resumetoken support. I'm wondering if I should just remove this test (and potentially replace with newer versions of mongodb)
EDIT: I added some logging in this [patch](https://spruce.mongodb.com/task/mongo_kafka_integration_tests__version~3.6_os~ubuntu_topology~replicaset_javaVersion~JDK17_integration_test_task_patch_d477f00c5506ac461ebc86cbeaaff291598d921c_67116e328067f60007286765_24_10_17_20_06_24?execution=0&sortBy=STATUS&sortDir=ASC) and found that for version 3.6 the cachedResumeToken is initially stored as null. It might be worth including a null check in the if condition for this logic to avoid these scenarios.

Moving this thread here. I agree with adding a null check. I don't expect customers to be using v3.6 in production, but it's still worth avoiding the null pointer exception. Out of curiosity, did you find any valuable details from the logs you've added?

Reply:
Yeah the [logs](https://parsley.mongodb.com/test/mongo_kafka_integration_tests__version~3.6_os~ubuntu_topology~replicaset_javaVersion~JDK17_integration_test_task_patch_d477f00c5506ac461ebc86cbeaaff291598d921c_67116e328067f60007286765_24_10_17_20_06_24/0/9f36feb7f14ab852d4ae3b9e0858a917?bookmarks=0,620&shareLine=0) showed that `cachedResumeToken` was being initialized as `null`. (search for `CALVIN:` if you wanted to review the logs)

I agree, I'll add a null check as well as an additional log so we can see if a null resume token is why we skipped that condition.

Reply:
Ok I just added the null check, I'll let the tests run but I do expect the test to still fail for version 3.6 because it doesn't have a resume token to set and resume from.

Reply:
Yep they fail in 3.6 for that reason still. https://spruce.mongodb.com/task/mongo_kafka_integration_tests__version~3.6_os~ubuntu_topology~replicaset_javaVersion~JDK17_integration_test_task_patch_d477f00c5506ac461ebc86cbeaaff291598d921c_671653eae106f80007f4ba73_24_10_21_13_15_23?execution=0&sortBy=STATUS&sortDir=ASC